### ISSUE TITLE
KeyboardDragHandler shouldn't rely on step

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/scenery-phet#336]
```KeyboardDragHandler``` uses ```step``` to manage key events, but it would be great if it didn't have to.
 
JavaScript doesn't natively handle multiple key presses at once.  When two keys are pressed down, the browser only gets ```keydown``` events for the second key pressed. ```KeyboardDragHandler``` gets around this by using ```keydown``` and ```keyup``` events to update a ```keystate``` object that internally tracks which keys are currently down.  ```step``` then updates a position Property  based on this ```keystate```.

But this requires that the drag handler always be stepped in the view, which is inconvenient. @zepumph and I think it would be great to brainstorm alternatives.

**Comment by zepumph** [comment]
I agree annoying, but the improved functionality that the current implementation has is well worth it! We should make sure that we don't sacrifice any of that to figure out a solution with less "step function parent passing" boiler plate. In the most recent application in faradays-law (see phetsims/faradays-law#80) I had to pass the step function from the screenView to a child, then to another child for the keyboardDragHandler to work. FL is a simple sim (relatively). I can imagine much worse. Perhaps other devs would have more insight on it as well.

**Comment by zepumph** [comment]
In https://github.com/phetsims/faradays-law/issues/89 I used KeyboardDragHandler as a guide to create a "sticky" drag handler for the FL magnet. That type relies on step as well. Whatever we do here should be carried over to that type.

**Comment by zepumph** [comment]
@jessegreenberg could you update this issue or close it. Does this still rely on step?

**Comment by jessegreenberg** [comment]
It no longer relies on step! Thanks @zepumph, closing.
